### ISSUE TITLE
XP today is not reliable

### ISSUE THREAD

**Issue by andreasscherbaum** [KartikTalwar/Duolingo#94]
I'm looking at two accounts, and see that "xp_today" is still set the next day, even though "streak_extended_today" is false, and it's already a new day.

Account 1:
```
lingo.get_streak_info()
{'daily_goal': 20, 'site_streak': xxx, 'streak_extended_today': False}

lingo.get_daily_xp_progress()
{   'lessons_today': [...],
    'xp_goal': 20,
    'xp_today': 94}
```

Account 2:
```
lingo.get_streak_info()
{'daily_goal': 10, 'site_streak': xxx, 'streak_extended_today': True}

lingo.get_daily_xp_progress()
{'lessons_today': [], 'xp_goal': 10, 'xp_today': 0}
```

Account 1 didn't do any lessons today, but the "lessons_today" list is populated, "xp_today" is set. However "streak_extended_today" is not set.

Account 2 did lessons today, however "xp_today" is 0, but "streak_extended_today" is set.

I'm aware that this is data coming from the Duolingo server, but I find that unreliable.
Is there a good way to get today's data and status?

**Comment by dmadisetti** [comment]
Yeah. Weird. Duolingo is reporting an timestamp in the future:
```json
{
    'xpGoal': 20,
    'xpGains': [{
        'eventType': 'LESSON',
        'xp': 12,
        'skillId': '4a5827801857c5210f6cea1066cda9c8',
        'time': 1602387744
    }, {
        'eventType': 'PRACTICE',
        'xp': 20,
        'skillId': None,
        'time': 1602388998
    }, {
        'eventType': 'PRACTICE',
        'xp': 15,
        'skillId': None,
        'time': 1602389306
    }, {
        'eventType': 'PRACTICE',
        'xp': 1 5,
        'skillId': None,
        'time': 1602390454
    }, {
        'eventType': 'PRACTICE',
        'xp': 15,
        'skillId': None,
        'time': 1602390669
    }],
    'streakData': {
        'length': 2,
        'updatedTim    eZone': 'America/New_York',
        'xpGoal': 20,
        'startTimestamp': 1602345600,
        'updatedTimestamp': 1602432000
    }
}
```
The relevant code for calculating updates is here, and it uses updatedTimestamp as reference. As of writing this it's `1602391856`, but `1602432000` (12PM GMT) is in the _future_.
This might be a datetime issue on their end, were you doing this near midnight your time?

An easy fix of course is just to use filter based on system clock datetime, see https://github.com/KartikTalwar/Duolingo/pull/96

**Comment by joshcoales** [comment]
I've merged #96 so that should close this
